File: Backend/src/schemas/news.py
from typing import Literal

from pydantic import BaseModel, Field


class NewsTeaserImage(BaseModel):
    imageVariants: dict[str, str] | None = None
    alttext: str | None = None


class NewsItem(BaseModel):
    sophoraId: str
    title: str
    topline: str | None = None
    firstSentence: str | None = None
    date: str | None = None
    shareURL: str | None = None
    detailsweb: str | None = None
    ressort: str | None = None
    breakingNews: bool | None = None
    teaserImage: NewsTeaserImage | None = None


class NewsResponse(BaseModel):
    news: list[NewsItem] = Field(default_factory=list)
    source: Literal["live", "cache"] = "live"

File: Frontend/nimrag-frontend/src/types/news.ts
import type { NewsRessort } from './appConfig'

export interface NewsTeaserImage {
  imageVariants?: Record<string, string> | null
  alttext?: string | null
}

export interface NewsItem {
  sophoraId: string
  title: string
  topline?: string | null
  firstSentence?: string | null
  date?: string | null
  shareURL?: string | null
  detailsweb?: string | null
  ressort?: string | null
  breakingNews?: boolean | null
  teaserImage?: NewsTeaserImage | null
}

export interface NewsQuery {
  ressort?: NewsRessort | null
  regions?: number[]
}

export interface NewsResponse {
  news: NewsItem[]
  source: 'live' | 'cache'
}